bytes-extended/src/main.sw

```diff
@@ -311,6 +311,18 @@ impl Bytes {
     }
 }
 
+impl Bytes {
+    pub fn with_ethereum_prefix(hash: b256) -> Self {
+        let prefix = "\x19Ethereum Signed Message:\n32";
```

Comment:
I'm realizing now I think I made a mistake in my `test_write_and_read_str` test that also applies here

If you try to get the length of `"\x19Ethereum Signed Message:\n"` when it's a rust str or javascript string, it gives a length of 26. But the sway compiler gives it a length of 30

I think it's because `\x19` is syntax for rust / javascript to say "this is one byte with the the hexadecimal values 1 and 9". But I think sway doesn't have this kind of escaping. Same goes for \n, Sway doesn't play well with it

I tested this to confirm and it looks like sway doesn't properly escape:

```
// These all pass
#[test()]
fn dissect_str() {
    let value = "\x19Ethereum Signed Message:\n";

    let mut bytes = Bytes::with_length(64);
    bytes.write_packed_bytes(0u64, __addr_of(value), 30);
    assert(bytes.read_u8(0) == 0x5cu8); // utf-8 encoded character \
    assert(bytes.read_u8(1) == 0x78u8); // utf-8 encoded character x
    assert(bytes.read_u8(2) == 0x31u8); // utf-8 encoded character 1
    assert(bytes.read_u8(3) == 0x39u8); // utf-8 encoded character 9

    assert(bytes.read_u8(28) == 0x5cu8); // utf-8 encoded character \
    assert(bytes.read_u8(29) == 0x6eu8); // utf-8 encoded character n
}
```

I can't really find how to get the functionality we need in a very native way, my impression is their string support is super weak. Maybe we should just do something like
```
// Write the 0x19
bytes.write_u8(0u64, 0x19u8);
let prefix = "Ethereum Signed Message:"; // len is 24
bytes.write_packed_bytes(1u64, __addr_of(prefix), 24u64);
// Write \n (0x0a is the utf-8 representation of \n)
bytes.write(25u64, 0x0a);
// etc
```

Obviously doesn't have to be exactly like that though, but just to illustrate manually writing the bytes we need.

This also makes me think we should create a `bytes-extended-test` package so we can test the behavior of this vs what's shipped in ethers-rs to ensure consistency, would you be up for adding that?

I just asked in their Discord to see if there's support for \x and \n specifically

Reply:
Doesn't seem like the str type can do this, got this from Mohammad
```
Some discussions here: https://github.com/FuelLabs/sway/issues/1104. The str type is likely not sufficient in its current form and not very ergonomic. Something that we're planning to address.
```